### DISCUSSION TITLE
OR Pseudo-classes

### DISCUSSION THREAD

**Discussion by emargareten** [tailwindlabs/tailwindcss#9116] [Ideas]
currently when I want to have a certain class applied to hover or focus etc. I have to write separate classes for each state:
```html
<button class="bg-sky-500 hover:bg-sky-700 focus:bg-sky-700 active:bg-sky-700 ...">
```

it would be great if we can have them all in a single class something like this:
```html
<button class="bg-sky-500 hover|focus|active:bg-sky-700 ...">
```

**Comment by zslabs** [comment]
Already covered in https://twitter.com/adamwathan/status/1461519820411789314

**Reply by emargareten** [reply]
@zslabs Isn't that for grouping together all classes for a certain state? my case here is for grouping states for a certain class.

**Reply by zslabs** [reply]
Is this maybe what you're looking for? https://tailwindcss.com/docs/plugins#adding-variants There's a `hocus` selector that you could extend to meet your needs

**Comment by moshest** [comment]
I saw the [variants plugin](https://tailwindcss.com/docs/plugins#adding-variants), maybe we can add it as native feature?

Syntax like `hover|focus|active:underline` or `hover,focus,active: underline` can work great.

**Reply by wongjn** [reply]
You could consider using [`:is`](https://developer.mozilla.org/en-US/docs/Web/CSS/:is) like `[&:is(:hover,:focus,:active)]:underline`.

**Reply by laserhybiz** [reply]
👍 for @wongjn's answer.

You can add it as a plugin in tailwind.config.js

```
plugins: [
  plugin(function({ matchVariant }) {
    matchVariant('is', (value) => `&:is(${value})`)
  })
]
```

then use it like this:
```
<div class="is-[:hover,:focus]:bg-blue-500 ...">
  <!-- ... -->
</div>
```
